src/main/export/export.ts

```diff
@@ -1,53 +1,71 @@
 /* eslint-disable no-plusplus */
 import { BrowserWindow } from 'electron';
-import { writeFileSync } from 'fs';
+import { existsSync, writeFileSync } from 'fs';
 import path, { join } from 'path';
-import { secondToTimestamp, padZeros } from '../timeUtils';
+import ffprobe from 'ffprobe';
+import ffprobeStatic from 'ffprobe-static';
+import { secondToEDLTimestamp, padZeros } from '../timeUtils';
 import { RuntimeProject, Transcription } from '../../sharedTypes';
 import { mkdir } from '../util';
 import convertTranscriptToCuts from '../../transcriptProcessing/transcriptToCuts';
+import { fracFpsToDec } from '../handlers/helpers/exportUtils';
 
 export const constructEDL: (
   title: string,
   transcription: Transcription,
   source: string | null,
   mainWindow: BrowserWindow | null
-) => string = (title, transcription, source, mainWindow) => {
-  let output = `TITLE: ${title}\nFCM: NON-DROP FRAME\n\n`;
+) => Promise<string> = async (title, transcription, source, mainWindow) => {
+  if (!source) {
+    throw Error('No Video Source');
+  } else {
+    let fps = 30; // default to a safe 30fps.
+    if (existsSync(source)) {
+      const videoData = await ffprobe(source, { path: ffprobeStatic.path });
+      fps = fracFpsToDec(videoData.streams[0].avg_frame_rate);
+    }
 
-  const cuts = convertTranscriptToCuts(transcription);
-  const entries = cuts.length;
+    let output = `TITLE: ${title}\nFCM: NON-DROP FRAME\n\n`;
+    // this can be technically incorrect in some rare cases, but it doesn't affect functionality
```

Comment:
What line does this comment apply to? If it's the above one then move it a line up

Reply:
Fixed!